sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -1627,8 +1627,8 @@ class DataFrame private[sql](
    * an RDD out to a parquet file, and then register that file as a table.  This "table" can then
    * be the target of an `insertInto`.
    *
-   * Also note that while this function can persist the table metadata into Hive's metastore,
-   * the table will NOT be accessible from Hive, until SPARK-7550 is resolved.
+   * Also note this function probably persist the table metadata into Hive's metastore, and we can
+   * disable that by setting spark.sql.hive.writeDataSourceSchema to false.
```

Comment:
This function *always* persist table metadata into Hive's metastore. But the table is not accessible from Hive unless the underlying data source is either Parquet or ORC.